redash/handlers/queries.py

```diff
@@ -50,7 +50,8 @@ def get(self):
 class QueryRecentAPI(BaseResource):
     @require_permission('view_query')
     def get(self):
-        recent = [d.to_dict() for d in models.Query.recent(current_user.id)]
+        recent = [d.to_dict() for d in models.Query.recent(
+                    limit_to_current_user=True)]
```

Comment:
While here you're limiting to the current user's groups, in the lines below we retrieve filler recents (in case the user doesn't have enough) where you don't apply this limit. You need to filter by groups there too...